atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/DelegatingTimelockService.java

```diff
@@ -16,39 +16,53 @@
 
 package com.palantir.timelock.paxos;
 
-import com.google.common.util.concurrent.ListenableFuture;
-import com.palantir.atlasdb.futures.AtlasFutures;
 import com.palantir.atlasdb.timelock.AsyncTimelockService;
-import com.palantir.lock.client.IdentifiedLockRequest;
+import com.palantir.lock.client.CommitTimestampGetter;
+import com.palantir.lock.client.LockLeaseService;
+import com.palantir.lock.client.TransactionStarter;
 import com.palantir.lock.v2.ClientLockingOptions;
-import com.palantir.lock.v2.IdentifiedTimeLockRequest;
 import com.palantir.lock.v2.LockImmutableTimestampResponse;
 import com.palantir.lock.v2.LockRequest;
 import com.palantir.lock.v2.LockResponse;
-import com.palantir.lock.v2.LockResponseV2;
 import com.palantir.lock.v2.LockToken;
-import com.palantir.lock.v2.RefreshLockResponseV2;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
 import com.palantir.lock.v2.TimelockService;
 import com.palantir.lock.v2.WaitForLocksRequest;
 import com.palantir.lock.v2.WaitForLocksResponse;
 import com.palantir.timestamp.TimestampRange;
+import java.util.List;
 import java.util.Set;
 
 final class DelegatingTimelockService implements TimelockService {
+    private final TransactionStarter transactionStarter;
+    private final LockLeaseService lockLeaseService;
     private final AsyncTimelockService timelock;
-
-    public DelegatingTimelockService(AsyncTimelockService timelock) {
+    private final CommitTimestampGetter commitTimestampGetter;
+
+    public DelegatingTimelockService(
+            TransactionStarter transactionStarter,
+            LockLeaseService lockLeaseService,
+            AsyncTimelockService timelock,
+            CommitTimestampGetter commitTimestampGetter) {
+        this.transactionStarter = transactionStarter;
+        this.lockLeaseService = lockLeaseService;
         this.timelock = timelock;
+        this.commitTimestampGetter = commitTimestampGetter;
+    }
+
+    @Override
+    public List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTransactionBatch(int count) {
+        return transactionStarter.startIdentifiedAtlasDbTransactionBatch(count);
     }
 
     @Override
     public long getFreshTimestamp() {
-        return timelock.getFreshTimestamp();
+        return getFreshTimestamps(1).getLowerBound();
```

Comment:
I'm curious why this change was necessary - sure, it might be that under the hood now, but do we guarantee that we will never change `getFreshTimestamp`? Probably, but may as well have it go through the same codepath as the rest.

Reply:
it was probably me trying to chase down some weird stuff in particular tests; I'll revert it and make sure that everything still works.